### PR TITLE
Added support for V1.3 interfaces

### PR DESCRIPTION
**Pull request by vbenni**
- Supports v1.3 interfaces except executeFenced
- Supports v1.0 and v1.1 interfaces.

Signed-Off-by: Vijeetkumar Benni <vijeetkumar.benni@intel.com>

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -65,6 +65,7 @@ shared_library("intel_nnhal") {
     "-fexceptions",
     "-fvisibility=default",
     "-DIMPLEMENT_INFERENCE_ENGINE_API",
+    "-DNN_DEBUG"
```
Comment (RavirajSitaram) [inline BUILD.gn:68]: Do not enable the debug flag by default.

File: BUILD.gn [hunk 2]
```diff
@@ -194,6 +223,7 @@ static_library("pugixml") {
     "-Wno-unused-private-field",
     "-Wno-invalid-partial-specialization",
     "-Wno-array-bounds",
+    "-Wno-implicit-fallthrough",
```
Comment (RavirajSitaram) [inline BUILD.gn:226]: After rebase, the implicit fallthrough messages should have been resolved.. Can you check again by removing this flag?

File: ModelManager.cpp [hunk 3]
```diff
@@ -26,22 +26,22 @@ bool NnapiModelInfo::updateOutputshapes(size_t outputIndex, std::vector<size_t>&
...
     }
     mOperands.resize(count);
-    mOutputShapes.resize(mModel.outputIndexes.size());
+    mOutputShapes.resize(mModel.main.outputIndexes.size());
 
     // Start by setting the runtime info to what's in the model.
-    for (size_t i = 0; i < count; i++) {
-        const Operand& from = mModel.operands[i];
-        dumpOperand(i, mModel);
+    for (int i = 0; i < count; i++) {
```
Comment (RavirajSitaram) [inline ModelManager.cpp:39]: this should be size_t else we will run into compiler warnings. Applies to all for loops modified in this PR

**Review by RavirajSitaram — CHANGES_REQUESTED** [review]

File: BasePreparedModel.cpp [hunk 4]
```diff
@@ -64,6 +65,33 @@ static Return<void> notify(const sp<V1_2::IExecutionCallback>& callback, const E
...
+                           const hidl_vec<OutputShape>& outputShapes, Timing timing) {
+    return callback->notify_1_3(convertToV1_3(status), outputShapes, timing);
+}
+
+static void floatToUint8(const float* src, uint8_t* dst, size_t size) {
+    for (uint32_t i = 0; i < size; ++i) {
+        dst[i] = static_cast<uint8_t>(src[i]);
+        ALOGV("%s input: %f output: %d ", __func__, src[i], dst[i]);
+    }
+}
```
Comment (anikulk) [inline BasePreparedModel.cpp:78]: can this live in utils.cpp/h?

**Comment by anikulk** [comment]
Any update here?

File: BasePreparedModel.cpp (hunk 4 above)
Comment (vbenni) [reply BasePreparedModel.cpp:78]: It is part of old code and my changes.
Looks like this is not used in any other place. So may not be needed to put in utils.h .

Comment (anikulk) [reply BasePreparedModel.cpp:78]: Okay makes sense.

**Review by anikulk — APPROVED** [review]

**Comment by vbenni** [comment]
> Any update here?

Resolved

**Review by RavirajSitaram — APPROVED** [review]
LGTM

**Comment by reaganlo** [comment]
@vbenni Hi Vijeet, does this PR pass the LayerNorm VTS test? 
E.g. **cros_nnapi_vts_1_3 --gtest_filter=*layer_norm_lstm_CifgPeepholeProjectionNoClippingLayerNormLstm**

**Comment by RavirajSitaram** [comment]
@reaganlo The aim for this PR is to enable 1.3 interfaces in Intel NNHAL. If there are specific test failures, let us work on them separately.

**Comment by reaganlo** [comment]
@RavirajSitaram @vbenni The tests actually pass. I was just using the incorrect version of OV. Ignore.

**Comment by vbenni** [comment]
> @vbenni Hi Vijeet, does this PR pass the LayerNorm VTS test?
> E.g. cros_nnapi_vts_1_3 --gtest_filter=*layer_norm_lstm_CifgPeepholeProjectionNoClippingLayerNormLstm

@reaganlo : All the tests related to LayerNorm are passing in v1.3

[----------] Global test environment tear-down
[==========] 66 tests from 6 test suites ran. (6520 ms total)
[  PASSED  ] 66 tests.

**Review by JeevakaPrabu — APPROVED** [review]

**Comment by Nagamani71** [comment]
@vbenni can you please run coding style

**Comment by vbenni** [comment]
> @vbenni can you please run coding style

Done

**Pull request merged**